src/object/cli_obj.c

```diff
@@ -4923,33 +4924,39 @@ obj_dup_sgls_free(struct obj_auxi_args *obj_auxi)
 
 				if (skip_sgl_iov(false, iov)) {
 					j++;
+					sg_nr_out++;
```

Comment:
Since each branch has `j++` and `sg_nr_out++`, then `j` and `sg_nr_out` are the same, why not only use `j` as following?

```
for (j = 0; j < sg_orig->sg_nr && dup_sg_idx < sg_dup->sg_nr_out; j++) {
...
}
sg_orig->sg_nr_out = j;

```
Then remove `j++` from each branch in the for section. That is more readable and clean.